docs/source/advanced/ui-tests.mdx

```diff
@@ -0,0 +1,32 @@
+---
+title: UI Tests
+---
+
+import {MultiCodeBlock} from 'gatsby-theme-apollo-docs';
+
+Apollo Android offers a built-in [IdlingResource](https://developer.android.com/reference/androidx/test/espresso/IdlingResource) to help writing UI tests with Espresso. The `ApolloIdlingResource` will make sure that your tests wait for your GraphQL queries terminate before moving on with testing.
+
+Add the `apollo-idling-resource` `dependency`:
+
+```kotlin:title=build.gradle[.kts]
+implementation("com.apollographql.apollo3:apollo-idling-resource:x.y.z")
+```
+
+> Note: If you have multiple `ApolloClients` you need to create and register multiple `ApolloIdlingResource` with different names. Registering several IdlingResources with the same name will crash.
```

Comment:
💡 Random idea: we could provide a default name like `ApolloIdlingResource-x` (x being a counter) .

Reply:
Let's see if that comes up as an issue before adding it?